src/tiledb/cloud/file/udfs.py

```diff
@@ -0,0 +1,88 @@
+from functools import partial
+from math import ceil
+from typing import Any, List, Mapping, Optional, Sequence, TypeVar
+
+from tiledb.cloud.file import utils as file_utils
+from tiledb.cloud.utilities import chunk
+from tiledb.cloud.utilities import find
+from tiledb.cloud.utilities import get_logger_wrapper
+
+DEFAULT_BATCH_SIZE = 100
+T = TypeVar("T")
+
+
+def find_uris_udf(
+    search_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    include: Optional[str] = None,
+    exclude: Optional[str] = None,
+    max_files: Optional[int] = None,
+    verbose: bool = False,
+) -> Sequence[str]:
+    """
+    Find URIs matching a pattern in the `search_uri` path.
+
+    `include` and `exclude` patterns are Unix shell style (see fnmatch module).
+
+    :param search_uri: URI to search for files
+    :param config: config dictionary, defaults to None
+    :param include: include pattern used in the search, defaults to None
+    :param exclude: exclude pattern applied to the search results, defaults to None
+    :param max_files: maximum number of URIs returned, defaults to None
+    :param verbose: verbose logging, defaults to False
+    :return: list of URIs
+    """
+    logger = get_logger_wrapper(verbose)
+
+    search_uri = search_uri.rstrip("/") + "/"
+    logger.info(f"Searching URI: {search_uri}...")
```

Comment:
For logging, we should use % formatting to avoid building the message when it's unused (and preserve better semantic info): `logger.info("Searching URI: %s", search_uri)`

Reply:
I was today years old when I found out about that for logging!
Will fix.